force-app/main/default/classes/CarTriggerUtility.cls

```diff
@@ -0,0 +1,63 @@
+public with sharing class CarTriggerUtility {
+    public static void afterInsert(List<Car__c> cars){
+        System.debug('utilitydebug');
+        List<Contact> listOfChangedContacts= new List<Contact>();
+        List<Contact> contacts = new List<Contact>();
+        
+        contacts= [SELECT Id,Expenses__c,LastAddedExpenseValue__c FROM Contact];
```

Comment:
Bardzo rzadko a zazwyczaj nigdy nie pozostawiamy Query bez Where, w momencie w którym masz powyżej 50k contactów w systemie to walniesz w limit a dodatkowo nigdy nie chcesz pętlić sie po wszystkim co masz w bazie danych.